### ISSUE TITLE
Can I build detectron2 on a nVidia Jetson Nano?

### ISSUE THREAD

**Issue by alexriedel1** [facebookresearch/detectron2#1424]
Hey, I want to build detectron2 on a nVidia Jetson Nano. The Jetson is running on Linux for Tegra so I was wondering if I can just build detectron2 the same way as on every other Linux system? Has anyone made some experiences doing this and can share it with me?

I don't want to use the detectron2 models in TensorRT or something, I just want plain detectron2 to run on the Jetson. 

Thanks!

**Comment by ppwwyyxx** [comment]
I'm not aware of any one that has tried this. But if all the dependencies can run there, detectron2 should be able to run there as well.

**Comment by Daneilsmoki** [comment]
Hi @alexriedel1 ,
can you explain me how you solved the problem? I also want to install detectron2 on my Nvidia Jetson Nano, but it's not working so far.

**Comment by alexriedel1** [comment]
> Hi @alexriedel1 ,
> can you explain me how you solved the problem? I also want to install detectron2 on my Nvidia Jetson Nano, but it's not working so far.

whats your issue?

**Comment by Daneilsmoki** [comment]
I have prepared a Nvidia Jetson with the following instructions: [https://developer.nvidia.com/embedded/learn/get-started-jetson-nano-devkit#intro](url)
Afterwards I created a new environment. Now I want to install detectron2 on this environment. Unfortunately all attempts have failed so far. I use Python 3.6.
Unfortunately it already fails with the installation of torchvision. I could still install torch 1.6.
Can you give me a step by step guide how you installed detectron2 on your Nvidia Jetson Nano please?
Detectron should run over the GPU.
Thanks a lot!

**Comment by alexriedel1** [comment]
Yes! You need to install pytorch for  aarch64 architectures, how to do this can be found here: https://forums.developer.nvidia.com/t/pytorch-for-jetson-nano-version-1-6-0-now-available/72048
I recommend pytorch 1.4.0 and torchvision 0.5.0

**Comment by Daneilsmoki** [comment]
Thanks for your answer. I installed torch and tried to test it, now the error message is: 
File "test.py", line 1, in <module>
   import torch
 File "home/ewa/vision/lib/python3.6/site-packages/torch/_init_.py", line 81, in <module>
  from torch._C import *
Import Error: libcudart.so.10.0: cannot open shared object file: No such file or directory

**Comment by alexriedel1** [comment]
one short google search later:
https://github.com/facebookresearch/detectron2/blob/master/INSTALL.md#common-installation-issues
" Undefined CUDA symbols; cannot open libcudart.so

The version of NVCC you use to build detectron2 or torchvision does not match the version of CUDA you are running with. This often happens when using anaconda's CUDA runtime.
Use python -m detectron2.utils.collect_env to find out inconsistent CUDA versions. In the output of this command, you should expect "Detectron2 CUDA Compiler", "CUDA_HOME", "PyTorch built with - CUDA" to contain cuda libraries of the same version.

When they are inconsistent, you need to either install a different build of PyTorch (or build by yourself) to match your local CUDA installation, or install a different version of CUDA to match PyTorch."

**Comment by Daneilsmoki** [comment]
I've already seen this clue but how can I find out what to install? 
The folder directory (usr/local) contains a folder cuda-10.2

**Comment by alexriedel1** [comment]
please tell you JetPack Version!
This site also tells you how to build from source, do it, building with you cuda version, which problably is 10.2
https://forums.developer.nvidia.com/t/pytorch-for-jetson-nano-version-1-6-0-now-available/72048

**Comment by Daneilsmoki** [comment]
I have already solved the problem. Thanks for your support!

**Comment by komenchuk** [comment]
Hello, could you give instructions on how to deploy Detectron2 to Jetson? How much FPS does your model give out?

**Comment by alexriedel1** [comment]
> Hello, could you give instructions on how to deploy Detectron2 to Jetson? How much FPS does your model give out?

Whats exactly your Problem when deploying as the instruction says? Check Pytorch Installation for ARM provide by nvidia, the link is above! FPS is very weak on jetson, maybe 0.2 FPS. On Xavier its about 3-4 FPS.

**Comment by cognitiveRobot** [comment]
@alexriedel1 why are you not using TensorRT for inference on Xavier? Is there any particular reason?

**Comment by alexriedel1** [comment]
I am using TensorRT,but wanted to try detectron2 just for testing. Also there is no way to convert the Faser RCNN RPN  to TensorRT if I am right?

**Comment by cognitiveRobot** [comment]
Yeah, that's true. we can't convert because of NMS of faster RCNN. so which framework are you using to train before deploying in TensorRT?

**Comment by alexriedel1** [comment]
> Yeah, that's true. we can't convert because of NMS of faster RCNN. so which framework are you using to train before deploying in TensorRT?

YOLOv4 :)

**Comment by komenchuk** [comment]
How much FPS gain does TensorRT give you? I trained the Mask-RCNN R50 FPN model, but I could not export it for TensorRT, what advice could you give me?

**Comment by alexriedel1** [comment]
I guess its gonna be some work either way to use mask rcnn on tensorrt. Check out this thread, I guess using the matterport tensorflow Implementation of mask rcnn is most promising to convert to tensorrt later https://forums.developer.nvidia.com/t/converting-mask-rcnn-to-tensor-rt/59956
